### ISSUE TITLE
Lean assertion violation on mathlib field_theory/intermediate_field.lean

### ISSUE THREAD

**Issue by timlacroix** [leanprover-community/lean#553]
### Prerequisites

* [ x] Put an X between the brackets on this line if you have done all of the following:
    * Checked that your issue isn't already [filed](https://github.com/leanprover-community/lean/issues).
        * Specifically, check out the [wishlist](https://github.com/leanprover-community/lean/issues?q=is%3Aissue+is%3Aopen+label%3AI-wishlist), open [RFCs](https://github.com/leanprover-community/lean/issues?q=is%3Aissue+is%3Aopen+label%3ARFC),
        or [feature requests](https://github.com/leanprover-community/lean/issues?q=is%3Aissue+is%3Aopen+label%3AFeature).
    * Reduced the issue to a self-contained, reproducible test case.

### Description

`lean mathlib/src/field_theory/intermediate_field.lean` leads to : 
```
LEAN ASSERTION VIOLATION
File: lean/src/frontends/lean/local_context_adapter.cpp
Line: 59
Task: mathlib/src/field_theory/intermediate_field.lean: parsing at line 48
!has_regular_local(r)
(C)ontinue, (A)bort/exit, (S)top/trap
```

**Reproduces how often:** 100%

### Versions

Fresh clone / debug build of lean + fresh mathlib clone.

**Comment by jcommelin** [comment]
I cannot reproduce this. Are you using `elan`? Might there be a version mismatch between the version of `lean` that you run and the version that mathlib is using?

**Comment by timlacroix** [comment]
This is without elan, on a debug build and setting LEAN_PATH manually to the right src files.
I can confirm that this issue does not appear in Release mode.

**Comment by gebner** [comment]
Yes, there are quite a few wrong assertions.  We just never notice them because they are not enabled in release mode, which everybody is using.  I think I'm just going to remove this assertion.

I'm also surprised that current mathlib master compiles with Lean master.  IIRC there were some breaking changes in e.g. fin.sub.

**Comment by gebner** [comment]
MWE:
```lean
set_option old_structure_cmd true

structure e :=
(w : ℕ)

structure a extends e :=
(x : ℕ)
(w := x)

structure b extends a
```

The error is somewhere in the structure command.  The code is not particularly nice, so I don't think I want to put effort into fixing this.

@timlacroix Does commenting out the assertion work for you?

**Comment by timlacroix** [comment]
@gebner yeah I can comment the assert, no problem.
